bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/experiencefragments/CommerceExperienceFragmentsRetriever.java

```diff
@@ -35,14 +35,14 @@ public interface CommerceExperienceFragmentsRetriever {
      * 
      * @return The a list of experience fragments that match this container.
      */
-    List<Resource> getExperienceFragmentsForProduct(String sku, String fragmentLocation, Page currentPage);
+    List<Resource> getExperienceFragmentsForProduct(String sku, String fragmentLocation, Integer maxItems, Page currentPage);
```

Comment:
For what is the `maxItems` parameter needed? All the usages we have use `1` for it.

Also the AssoicatedContentService could replace this service completely.